src/hooks/useLoadDocuments.ts

```diff
@@ -29,11 +29,16 @@ const formatServerResponse = (documents: DocumentFromServer[]): Document[] =>
 
 export const loadDocuments = async (discipline: Discipline): Promise<Document[]> => {
   const url = ENDPOINTS.documents.replace(':id', `${discipline.id}`)
-
   const response = await getFromEndpoint<DocumentFromServer[]>(url, discipline.apiKey)
   return formatServerResponse(response)
 }
 
+export const loadDocumentsById = async (disciplineId: number): Promise<Document[]> => {
```

Comment:
This is quite redundant and duplicated. Should be better to refactor `loadDocuments` to only take `disciplineId` and optionally an `apiKey`. Then you can either change `useLoadDocuments` such that it also just takes an id + apiKey or just pass discipline.id and discipline.apiKey as params

Reply:
You are right, i tried this, but I couldn't figure out how to pass multiple params to useLoadAsync.

`
const useLoadDocuments = (disciplineId: number, apiKey?: string): Return<Document[]> => useLoadAsync(loadDocuments, {discipline.id, discipline.apiKey})
`

I get `TS2345: Argument of type '(disciplineId: number, apiKey: string) => Promise ' is not assignable to parameter of type '(params: number) => Promise '`

Reply:
You just use an object as well, i.e.:
``` ts
const const useLoadDocuments = ({ id, apiKey }: { id: number, apiKey?: string }): Return<Document[]> => useLoadAsync(loadDocuments, { id, apiKey })
```